src/ActionList2/Item.tsx

```diff
@@ -0,0 +1,197 @@
+import {useSSRSafeId} from '@react-aria/ssr'
+import {useColorSchemeVar, useTheme} from '../ThemeProvider'
+import Box, {BoxProps} from '../Box'
+import {get} from '../constants'
+import {SxProp, merge} from '../sx'
+import createSlots from '../utils/create-slots'
+import {AriaRole} from '../utils/types'
+import {ListContext} from './List'
+import {customItemThemes} from './hacks'
+import {Selection} from './Selection'
+
+export const getVariantStyles = (variant: ItemProps['variant'], disabled: ItemProps['disabled']) => {
+  if (disabled) {
+    return {
+      color: 'fg.muted',
+      iconColor: 'fg.muted',
+      annotationColor: 'fg.muted'
+    }
+  }
+
+  switch (variant) {
+    case 'danger':
+      return {
+        color: 'danger.fg',
+        iconColor: 'danger.fg',
+        annotationColor: 'fg.muted'
+      }
+    default:
+      return {
+        color: 'fg.default',
+        iconColor: 'fg.muted',
+        annotationColor: 'fg.muted'
+      }
+  }
+}
+
+export type ItemProps = {
+  /**
+   * Primary content for an Item
+   */
+  children: React.ReactNode
+  /**
+   * Callback that will trigger both on click selection and keyboard selection.
+   */
+  onSelect?: (event: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>) => void
+  /**
+   * Is the `Item` is currently selected?
+   */
+  selected?: boolean
+  /**
+   * Style variations associated with various `Item` types.
+   *
+   * - `"default"` - An action `Item`.
+   * - `"danger"` - A destructive action `Item`.
+   */
+  variant?: 'default' | 'danger'
+  /**
+   * Items that are disabled can not be clicked, selected, or navigated through.
+   */
+  disabled?: boolean
+  /**
+   * The ARIA role describing the function of `Item` component. `option` is a common value. |
+   */
+  role?: AriaRole
+} & SxProp
+
+const {Slots, Slot} = createSlots(['LeadingVisual', 'InlineDescription', 'BlockDescription', 'TrailingVisual'])
+export {Slot}
+export type ItemContext = Pick<ItemProps, 'variant' | 'disabled'> & {
+  inlineDescriptionId: string
+  blockDescriptionId: string
+}
+
+export const Item = React.forwardRef<HTMLLIElement, ItemProps>(
+  (
+    {variant = 'default', disabled = false, selected = undefined, onSelect = () => null, sx = {}, ...props},
+    forwardedRef
+  ): JSX.Element => {
+    const customItemTheme = customItemThemes[variant]
+    const {variant: listVariant, showDividers} = React.useContext(ListContext)
+
+    const {theme} = useTheme()
+
+    const styles = {
+      display: 'flex',
+      paddingX: get('space.2'),
+      paddingY: '6px',
+      marginX: listVariant === 'inset' ? get('space.2') : 0,
+      minHeight: get('space.5'),
+      borderRadius: get('radii.2'),
+      transition: 'background 33.333ms linear',
+      color: getVariantStyles(variant, disabled).color,
+      textDecoration: 'none', // for as="a"
+
+      ':not([aria-disabled])': {cursor: 'pointer'},
```

Comment:
No need because we set aria-disabled ourselves :)

https://github.com/primer/react/pull/1517/files/a86b13b99321cb5b43159c96cb4b17f016e80b8d#diff-6b3f2a26879573bcf7068c2c16107ae50982e6aff5d735478c13ca231dd8e22bR152